compiler/src/dotty/tools/dotc/transform/Bridges.scala

```diff
@@ -43,6 +42,25 @@ class Bridges(root: ClassSymbol, thisPhase: DenotTransformer)(using Context) {
       OverridingPairs.isOverridingPair(sym1, sym2, parent.thisType)
   }
 
+  /** In general we don't want to create bridges for methods defined in traits, but for some cases it is necessary,
+   *  especially for safe invocation of methods inside SAMs (see issue #15402).
+   *  In same cases creation of bridge inside trait to a method with erased generic method might lead to incorrect
+   *  interface method lookup and infinite loops at runtime. (eg. cats, WriterTApplicative.map)
+   *  Becouse of that issue, we limit bridges to only methods with the same set of paramaters and a different, covariant, result type.
+   *  When adding bridges to traits ignore non-public methods (see `DottyBackendTests.invocationReceivers` test case)
```

Comment:
```suggestion
  /** Usually, we don't want to create bridges for methods defined in traits, but for some cases it is necessary.
   *  For example with SAM methods combined with covariant result types (see issue #15402).
   *  In some cases, creating a bridge inside a trait to an erased generic method leads to incorrect
   *  interface method lookup and infinite loops at run-time. (e.g., in cats' `WriterTApplicative.map`).
   *  To avoid that issue, we limit bridges to methods with the same set of parameters and a different, covariant result type.
   *  We also ignore non-public methods (see `DottyBackendTests.invocationReceivers` for a test case).
```
This explains why the restriction is necessary, but it doesn't justify why it's fine. In other words, with the restriction, could we still run into other issues similar to #15402 ?